TeachingRecordSystem/src/TeachingRecordSystem.SupportUi/Pages/Alerts/Alert/Index.cshtml.cs

```diff
@@ -1,10 +1,74 @@
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.AspNetCore.Mvc.Filters;
 using Microsoft.AspNetCore.Mvc.RazorPages;
+using TeachingRecordSystem.Core.Dqt.Models;
+using TeachingRecordSystem.Core.Dqt.Queries;
+using TeachingRecordSystem.SupportUi.Pages.Common;
 
 namespace TeachingRecordSystem.SupportUi.Pages.Alerts.Alert;
 
 public class IndexModel : PageModel
 {
-    public void OnGet()
+    private readonly TrsLinkGenerator _linkGenerator;
+    private readonly ICrmQueryDispatcher _crmQueryDispatcher;
+
+    public IndexModel(
+        TrsLinkGenerator linkGenerator,
+        ICrmQueryDispatcher crmQueryDispatcher)
+    {
+        _linkGenerator = linkGenerator;
+        _crmQueryDispatcher = crmQueryDispatcher;
+    }
+
+    [FromRoute]
+    public Guid AlertId { get; set; }
+
+    public AlertInfo? Alert { get; set; }
+
+    public bool IsActive { get; set; }
+
+    public Guid? PersonId { get; set; }
+
+    public async Task<IActionResult> OnPost()
```

Comment:
I think we should have two separate, named handler methods for activate and reactivate.

Reply:
Done